docs(toolbar): Update Toolbar setup docs (#12566)
getsentry/sentry-docs: docs/product/dev-toolbar/setup.mdx

  [GitHub](https://github.com/getsentry/sentry-toolbar/issues) if you have any feedback or concerns.
</Alert>

## Pre-requisites

For the Sentry toolbar to work best [enable tracing](/platforms/javascript/tracing/) in your app. With tracing enabled the Dev Toolbar will be able to associate issues and feedback with the current url in the browser location.

> ## Install

Reviewer: The steps to set up the Dev Toolbar are implied when you say "**The first step** is to include the toolbar code in your app." and "Next you will need to edit..."
but I think for easier digestion of the onboarding it would be nice if the steps were enumerated in some way?

It's also unclear which steps are optional v. required (for example, enabling FF is _technically_ optional as an organization may not use them in their development)

Author: I'll put numerals before the first three steps:
![SCR-20250204-mkfb](https://github.com/user-attachments/assets/527ebac0-a1ec-4c95-b764-a46ddfdcf41a)

Author: updated again. see the description for the latest screenshot of the outline / right sidebar.

> The first step is to include the toolbar code in your app.

Reviewer: ```suggestion
The first step is to include the Toolbar code in your app.
```

Reviewer: I won't edit the rest of the 'toolbar' mentions -- I'll let you decide if you want to keep the capitalization 😅 I think it makes sense since this is referring to our product Dev Toolbar and not a generic toolbar (say, the one in screenshot annotaitons)

> Since the Dev Toolbar will be visible to users within your app it's strongly recommended that you consider which environments the toolbar should apply to. See [Deploying to Different Environments](#deploying-to-different-environments) below for more.

Reviewer 2: ```suggestion
Since the Dev Toolbar will be visible to users within your app, it's strongly recommended that you consider which environments the toolbar should apply to. See [Deploying to Different Environments](#deploying-to-different-environments) below for more.
```

> Since the Dev Toolbar will be visible to users within your app it's strongly recommended that you consider which environments the toolbar should apply to. See [Deploying to Different Environments](#deploying-to-different-environments) below for more.

Reviewer: I'm struggling to understand the differences between "Deploying to Different Environments" and "Allow Domains" - it seems like both help you to zero-in on particular environments?

It's odd to me that we say "Deploying to Different Environments" is such a critical step but it falls second-last in the 'steps' 🤔

Author: i'll gonna re-do it all again, move the 'different envs' stuff in between step 1 and 2... maybe there'll be 4 steps now

```html {tabTitle: CDN}
<!--
Put this at the bottom of your page
so it doesn’t block other critical JavaScript.
-->
<script src="https://browser.sentry-cdn.com/sentry-toolbar/latest/toolbar.min.js"></script>
```
```typescript {tabTitle: React}
// An NPM package is under development

> // In the meantime, go here for instructions to create a react hook manually:

Reviewer 2: ```suggestion
// In the meantime, go here for instructions to create a React hook manually:
```

//   https://github.com/getsentry/sentry-toolbar/blob/main/docs/conditional-script.md
```

## Allow Domains

> Next you will need to edit the [Project Settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/projects/) page to allow the toolbar to connect to Sentry by configuring your dev, staging and production domains. Only add domains to this list that you trust and control.

Reviewer 2: ```suggestion
Next you will need to edit the [Project Settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/projects/) page to allow the toolbar to connect to Sentry by configuring your dev, staging, and production domains. Only add domains to this list that you trust and control.
```

> Next you will need to edit the [Project Settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/projects/) page to allow the toolbar to connect to Sentry by configuring your dev, staging and production domains. Only add domains to this list that you trust and control.

Reviewer 3: ```suggestion
Next you will need to edit the [Project Settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/projects/) page to allow the toolbar to connect to Sentry by configuring your dev, staging, and production domains. Only add domains to this list that you trust and control.
```

![Sentry's Dev Toolbar Settings Page](./img/sentry-project-settings-toolbar.png)

## Configure

Finally, call `SentryToolbar.init(initConfig)` to render a toolbar instance on each page where you want to see the Dev Toolbar. This will prompt any visitor to the page to login to your Sentry organization.

```html {tabTitle: CDN}
<html>
<head>...</head>
<body>
    ...
    <script src="https://browser.sentry-cdn.com/sentry-toolbar/latest/toolbar.min.js"></script>
    <script>
    window.SentryToolbar.init({ ... });
    </script>
</body>
</html>
```
```typescript {tabTitle:React}
// An NPM package is under development

> // In the meantime, go here for instructions to create a react hook manually:

Reviewer 2: ```suggestion
// In the meantime, go here for instructions to create a React hook manually:
```

  [GitHub](https://github.com/getsentry/sentry-toolbar/issues) if you have any feedback or concerns.
</Alert>

> ## Pre-requisites

Reviewer 4: ```suggestion
## Pre-Requisites
```

> For the Sentry Toolbar to work best, [enable tracing](/platforms/javascript/tracing/) in your app. With tracing enabled, the Dev Toolbar will be able to associate issues and feedback with the current URL in the browser location.

Reviewer 4: ```suggestion
For the Sentry Dev Toolbar to work best, [enable tracing](/platforms/javascript/tracing/) in your app. With tracing enabled, the Dev Toolbar will be able to associate issues and feedback with the current URL in the browser location.
```

## 1. Choose Deploy Environments

> Since the Dev Toolbar will be visible to users within your app, it's important to consider which environments the Toolbar should apply to.

Reviewer 4: ```suggestion
Since the Dev Toolbar will be visible to users within your app, it's important to consider which environments should display it.
```

> In dev & staging environments, include the Toolbar so that all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.

Reviewer 4: ```suggestion
In dev and staging environments, include the Toolbar so that all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.
```

In production environments, the Dev Toolbar can make it easier to reproduce real issues your users are having. However the Toolbar should not be rendered for all users of the site -- only when an employee/engineer/etc visits.

Once you decide where and when you want the Toolbar to appear, you'll write those conditions into your codebase. The specific implementation is something you'll need to write based on how your app works and how your dev team is set up.

For example, the conditions to show show the Toolbar in dev and staging might look like this:

```typescript
const env = process.env.SENTRY_ENVIRONMENT || 'development';

const isDev = env === 'development' || env === 'staging';
if (isDev) {
  // Enable the Dev Toolbar here...
}
```

Or if your web application requires authentication to access, you could add a conditional where the Dev Toolbar is shown always when deployed to development **and** staging, but in production only show the Toolbar **if** an employee is logged in.

## 2. Allow Domains

> You will need to edit the [Project Settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/projects/) page to allow the Toolbar to connect to Sentry by configuring your dev, staging, and production domains. Only add domains to this list that you trust and control.

Reviewer 4: ```suggestion
You will need to edit the [Project Settings](https://sentry.io/orgredirect/organizations/:orgslug/settings/projects/) page to allow the Toolbar to connect to Sentry by configuring your dev, staging, and production domains. Only add domains that you trust and control to this list.
```

![Sentry's Dev Toolbar Settings Page](./img/sentry-project-settings-toolbar.png)

## 3. Install

> Next you must include the Toolbar code in your app.

Reviewer 4: ```suggestion
Next you must include the Toolbar code in your app:
```

| `domId (optional)` | `string \| undefined` | The `id` given to the \<div\> that is created to contain the toolbar html. | `'sentry-toolbar'` |
| `debug (optional)` | `string \| undefined` | A comma separated string of debug targets to enable. Example: `'logging,state'`. If the list contains 'all' or 'true' then all targets will be enabled. Valid targets: `'logging' 'login-success' 'settings' 'state'` | `undefined` |
| `mountPoint (optional)` | `HTMLElement \| () => HTMLElement \| undefined` | Where to mount the toolbar in the DOM. | `document.body` |

### Unmounting The Toolbar

If you have called `SentryToolbar.init({...})` to render the toolbar, but now want to manually remove or unmount it from the page, you can call the cleanup function that is returned from `init()`. This will unmount all the injected HTML and CSS. Login credentials will not be removed, so you can re-insert the toolbar and still be authenticated.
```javascript
const unmountToolbar = window.SentryToolbar.init({ ... });

// sometime later...
unmountToolbar();
```

## Deploying to Different Environments

> Since the Dev Toolbar will be visible to users within your app it's strongly recommended that you consider which environments the toolbar should apply to.

Reviewer 2: ```suggestion
Since the Dev Toolbar will be visible to users within your app, it's strongly recommended that you consider which environments the toolbar should apply to.
```

> In dev and staging environments include the toolbar so that all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.

Reviewer 2: ```suggestion
In dev and staging environments, include the toolbar so that all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.
```

> In dev and staging environments include the toolbar so that all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.

Reviewer 3: ```suggestion
In dev and staging environments, include the toolbar so that all developers and testers can use it and quickly go from the page they're looking at back to Sentry for further debugging.
```

> In producion environments the Dev Toolbar can make it easier to reproduce real issues your users are having, but the toolbar should not be rendered for all users of the site, only when an employee/engineer/etc visits.

Reviewer 2: ```suggestion
In producion environments, the Dev Toolbar can make it easier to reproduce real issues your users are having, but the toolbar should not be rendered for all users of the site -- only when an employee/engineer/etc visits.
```

> In producion environments the Dev Toolbar can make it easier to reproduce real issues your users are having, but the toolbar should not be rendered for all users of the site, only when an employee/engineer/etc visits.

Reviewer 3: ```suggestion
In production environments the Dev Toolbar can make it easier to reproduce real issues your users are having, but the toolbar should not be rendered for all users of the site, only when an employee/engineer/etc visits.
```

The specific implementation for conditionally including the Dev Toolbar is something you'll need to write based on how your app works and how your dev team is set up. For example, the conditions to show show the toolbar in dev and staging might look like this:
```typescript
const env = process.env.SENTRY_ENVIRONMENT || 'development';

const isDev = env === 'development' || env === 'staging';
if (isDev) {
  window.SentryToolbar.init({ ... });
}
```

Or if your web application requires authentication to access, you could add a conditional where the Dev Toolbar is shown always when deployed to development **and** staging, but in production only show the toolbar **if** an employee is logged in.

If the toolbar `<script>` is accidentally included on your site, and `SentryToolbar.init()` is called, then a "Login to Sentry" button will be visible to the public. This is not ideal, but your data in Sentry will still be safe as users outside of your Sentry organization will not be able to authenticate themselves.

> ## Feature Flag Adapters

Reviewer: Should we make this a more straight forward name like "Setting Up the Feature Flag Panel"? rather than referring the adapter technology

Author: I'm going to also link back to the FF issues stuff

Author: it's "Feature Flag Panel" now. 

And added an info:
>If you're using feature flags inside your product then also setup [Feature Flag Evaluation and Change Tracking](https://docs.sentry.io/platforms/javascript/feature-flags/) within your SDK.

> In order to integrate your feature flagging platform with the Dev Toolbar you will need an adapter that can read flag data from your provider. It will also store and retrieve a list of overrides to apply to your local browser session.

Reviewer 2: ```suggestion
In order to integrate your feature flagging platform with the Dev Toolbar, you will need an adapter that can read flag data from your provider. It will also store and retrieve a list of overrides to apply to your local browser session.
```

> In order to integrate your feature flagging platform with the Dev Toolbar you will need an adapter that can read flag data from your provider. It will also store and retrieve a list of overrides to apply to your local browser session.

Reviewer 3: ```suggestion
In order to integrate your feature flagging platform with the Dev Toolbar, you will need an adapter that can read flag data from your provider. It will also store and retrieve a list of overrides to apply to your local browser session.
```

> There is a built-in `OpenFeatureAdapter` that is compatible with the [open-feature/js-sdk-contrib](https://github.com/open-feature/js-sdk-contrib). To use it call the `SentryToolbar.OpenFeatureAdapter` implementation.

Reviewer 2: ```suggestion
There is a built-in `OpenFeatureAdapter` that is compatible with the [open-feature/js-sdk-contrib](https://github.com/open-feature/js-sdk-contrib). To use it, call the `SentryToolbar.OpenFeatureAdapter` implementation.
```

```html {tabTitle: CDN}
<script>
// Define your provider
const provider = new FlagdWebProvider({...});

// Set the provider into the OpenFeature SDK
OpenFeature.setProvider(provider);

window.SentryToolbar.init({
  ...
  // Set the provider into the OpenFeatureAdapter
  featureFlags: window.SentryToolbar.OpenFeatureAdapter({provider})
});
</script>
```
```typescript {tabTitle: React}
import {OpenFeature} from '@openfeature/web-sdk';
import {FlagdWebProvider} from '@openfeature/flagd-web-provider';

// Define your provider
const provider = new FlagdWebProvider({...});

// Set the provider into the OpenFeature SDK
OpenFeature.setProvider(provider);

window.SentryToolbar.init({
  ...
  // Set the provider into the OpenFeatureAdapter
  featureFlags: window.SentryToolbar.OpenFeatureAdapter({provider})
});
```

> You can also create your own adapter by implemting the [`FeatureFlagAdapter` interface](https://github.com/getsentry/sentry-toolbar/blob/main/src/lib/types/featureFlags.ts).

Reviewer 2: ```suggestion
You can also create your own adapter by implementing the [`FeatureFlagAdapter` interface](https://github.com/getsentry/sentry-toolbar/blob/main/src/lib/types/featureFlags.ts).
```

| `placement (optional)` | `'right-edge' \| 'bottom-right-corner'` | Where to render the Toolbar on the screen. | `'right-edge'` |
| `theme (optional)` | `'system' \| 'dark' \| 'light'` | Whether to use dark or light mode. | `'system'` |
| `featureFlags (optional)` | `FeatureFlagAdapter \| undefined` | See [Feature Flag Panel](/product/dev-toolbar/setup//#feature-flag-Panel) below | `undefined` |
| `sentryOrigin (optional)` | `string \| undefined` | The origin where Sentry can be found. Used for loading the connection to Sentry, and generating links to the website. For example: `'https://acme.sentry.io'` | `'https://sentry.io'` |
| `domId (optional)` | `string \| undefined` | The `id` given to the \<div\> that is created to contain the Toolbar html. | `'sentry-toolbar'` |
| `debug (optional)` | `string \| undefined` | A comma separated string of debug targets to enable. Example: `'logging,state'`. If the list contains 'all' or 'true' then all targets will be enabled. Valid targets: `'logging' 'login-success' 'settings' 'state'` | `undefined` |
| `mountPoint (optional)` | `HTMLElement \| () => HTMLElement \| undefined` | Where to mount the Toolbar in the DOM. | `document.body` |

> ### Unmounting The Toolbar

Reviewer 4: ```suggestion
### Unmounting the Toolbar
```

If you have called `SentryToolbar.init({...})` to render the Toolbar, but now want to manually remove or unmount it from the page, you can call the cleanup function that is returned from `init()`. This will unmount all the injected HTML and CSS. Login credentials will not be removed, so you can re-insert the toolbar and still be authenticated.
```javascript
const unmountToolbar = window.SentryToolbar.init({ ... });

// sometime later...
unmountToolbar();
```

## Feature Flag Panel

<Alert>

>   If you're using feature flags inside your product then also setup [Feature Flag Evaluation and Change Tracking](https://docs.sentry.io/platforms/javascript/feature-flags/) within your SDK.

Reviewer 4: ```suggestion
  If you're using feature flags inside your product then also setup [Feature Flag Evaluation and Change Tracking](/platforms/javascript/feature-flags/) within your SDK.
```
